system-parachains/coretime/coretime-polkadot/src/migrations.rs

```diff
@@ -0,0 +1,330 @@
+
+		// Add the system pool core to the workplan starting now.
+		let now = 287000; // TODO - this needs to be after the cores have just added have been processed, which takes
+				  // two sessions. Currently just a placeholder. This part is probably better as a call to
+				  // assign_core on the relay as part of the referendum instead of here.
+		Workplan::<Runtime>::insert((now, 61), pool_assignment);
+
+		// TODO finalise the weights here.
+		<Runtime as frame_system::Config>::DbWeight::get()
+			.writes(1)
+			.saturating_mul(LEASES.len() as u64)
+			.saturating_add(BrokerWeights::request_core_count(62))
+			.saturating_add(<Runtime as frame_system::Config>::DbWeight::get().reads(1))
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+		// Idempotency check - this core will never be renewable at this timeslice ever again.
+		if PotentialRenewals::<Runtime>::get(PotentialRenewalId { core: 6, when: 292605 }).is_none()
+		{
+			return Ok(Vec::new())
+		}
+		let sale_info = SaleInfo::<Runtime>::get().unwrap();
+		let leases = Leases::<Runtime>::get();
+		let pre_upgrade_state = (sale_info, leases);
+		Ok(pre_upgrade_state.encode())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(state: Vec<u8>) -> Result<(), TryRuntimeError> {
+		if state.is_empty() {
+			return Ok(())
+		}
+		let (prev_sale_info, prev_leases): (SaleInfoRecordOf<Runtime>, LeasesRecordOf<Runtime>) =
+			Decode::decode(&mut &state[..]).unwrap();
+
+		log::info!(target: TARGET, "Checking migration.");
+
+		let sale_info = SaleInfo::<Runtime>::get().unwrap();
+
+		// Check the sale start has not changed.
+		assert_eq!(sale_info, prev_sale_info);
+
+		// The workplan entries start from the region begin reported by the new SaleInfo.
+		let workplan_start = sale_info.region_begin;
+
+		// Check the reservations are still in the workplan out of an abundance of caution.
+		for (core_id, task) in [Task(1001), Task(1002), Task(1000), Task(1004), Task(1005)]
+			.into_iter()
+			.enumerate()
+		{
+			assert_eq!(
+				Workplan::<Runtime>::get((workplan_start, core_id as u16)),
+				Some(Schedule::truncate_from(Vec::from([ScheduleItem {
+					mask: CoreMask::complete(),
+					assignment: task,
+				}])))
+			);
+		}
+
+		// Make sure we've got all the leases.
+		let leases = Leases::<Runtime>::get();
+		assert_eq!(leases.len(), LEASES.iter().filter(|(_, l)| sale_info.region_end <= *l).count());
+
+		// Make something out of the leases that is easier to check against.
+		let leases_vec: Vec<(u32, u32)> =
+			leases.iter().map(|LeaseRecordItem { until, task }| (*task, *until)).collect();
+
+		// Iterate through hardcoded leases and check they're all correctly in state and scheduled
+		// in the workplan.
+		for (i, (para_id, until)) in LEASES.iter().enumerate() {
+			// Add the system parachains as an offset - these should come before the leases.
+			let core_id = i as u16 + 4;
```

Comment:
```suggestion
			let core_id = i as u16 + 5;
```
There are 5 system chains, I believe the offset here should be 5.